src/rand.rs

```diff
@@ -195,6 +205,21 @@ mod sysrand_or_urandom {
     }
 }
 
+#[cfg(any(target_os = "macos", target_os = "ios"))]
+mod security {
+    use error;
+
+    pub fn fill(dest: &mut [u8]) -> Result<(), error::Unspecified> {
+        let r = unsafe {
+            super::SecRandomCopyBytes(super::kSecRandomDefault, dest.len(), dest.as_mut_ptr())
+        };
+        match r {
+            0 => Ok(()),
+            _ => Err(error::Unspecified),
+        }
+    }
```

Comment:
@briansmith: So I was originally doing something like this:

```rs
#[repr(c)]
struct SecRandomRef {}
```

But the compiler complained about empty struct layout (for good reason). I'm guessing `enum`—as you suggested—is the proper way to do this?

(Just satisfying my curiosity here. 🙃 )